### PR TITLE
fix: Use smarter quoting for ProxyCommand in config-ssh

### LINKED ISSUE #2853: Windows: ssh coder.workspace not working after running coder config-ssh
Windows 10
Coder version: 0.7.6

Steps:
1) create a workspace
2) run coder config-ssh
3) run ssh coder.workspace

**Actual Result:** 
CreateProcessW failed error:87
posix_spawn: Unknown error
2022-07-08_18h33_34
**Expected Result:** connected successfully

4) correct .ssh file: remove double quotes and double slashes from paths in ProxyCommand
5) run ssh coder.workspace

**Actual=Expected Result:** connected successfully
2022-07-08_18h34_44

User has to update .ssh file every time after running coder config-ssh to be able to connect via ssh coder.workspace

### PR DESCRIPTION
**Pull request by mafredri**
This change takes better into account how OpenSSH executes
`ProxyCommand`s and applies quoting accordingly.

This supercedes #3664, which was reverted.

Fixes #2853

---

Bonus commit:

- fix: Ensure `~/.ssh` directory exists

### CODE REVIEW

File: cli/configssh.go [hunk 1]
```diff
@@ -482,6 +502,50 @@ func writeWithTempFileAndMove(path string, r io.Reader) (err error) {
...
+	if strings.Contains(path, "\"") {
+		path = strings.ReplaceAll(path, "\"", "\\\"")
+	}
+	// A space or a tab requires quoting, but tabs must not be escaped
+	// (\t) since OpenSSH interprets it as a literal \t, not a tab.
+	if strings.ContainsAny(path, " \t") {
+		path = fmt.Sprintf("\"%s\"", path)
+	}
+	return path, nil
+}
```
Comment (johnstcn) [inline cli/configssh.go:547]: So --- and hear me out here 😳 --- could we test this by creating a hello-world batch or shell script (platform-dependent) under a temp path that contains a space, attempting to execute `/bin/bash -c '<path of thing>'`, and validating that it executes successfully?

Comment (mafredri) [reply cli/configssh.go]: I don't see why not, good idea! In theory, we'd want to test all the shells.. but then again, I think we're unlikely to run into that many edge cases with this unless people do really fringe stuff with their filesystems/folder structures. 😅

Comment (johnstcn) [reply cli/configssh.go]: I think testing `/bin/sh` and/or `cmd.exe` is all we should do tbh

**Review by johnstcn — APPROVED** [review]
🚀

**Pull request merged**